### PR TITLE
stage control

### PR DESCRIPTION
**Pull request by AngleQian**

### CODE REVIEW

File: src/Extension.Tests/StageControlTests.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            return new Stage(stageId, Enumerable.Empty<string>());
+        }
+
+        [Fact]
+        public void stage_with_no_dependency_is_auto_revealed_single_stage_case()
+        {
+            List<int> revealedStageIds = new List<int>();
+
+            var stageController = new StageController();
+            stageController.AddBlankStages(1);
```
Comment (jonsequitur) [inline src/Extension.Tests/StageControlTests.cs:33]: Why ints instead of objects?

Comment (AngleQian) [reply src/Extension.Tests/StageControlTests.cs:33]: It is a stageId. StageIds are ints at the moment. But I could make them into more complex types?

Comment (AngleQian) [reply src/Extension.Tests/StageControlTests.cs:33]: There is a chance stageId's might be entirely internal, which means we probably won't need stageIds. The teacher sees stage names (strings) and those are mapped to stage objects directly.

Comment (jonsequitur) [reply src/Extension.Tests/StageControlTests.cs:33]: IDs are useful for tracking identities in external systems (services, databases) but less so when you already have object references. In fact the ID and the reference can get out of sync, so they can be a bit of a liability when everything is in-memory.

Comment (AngleQian) [reply src/Extension.Tests/StageControlTests.cs]: makes sense. I've removed ids

**Pull request merged**